jinahub/encoders/text/CLIPTextEncoder/clip_text.py

```diff
@@ -1,69 +1,117 @@
-        default_traversal_paths: List[str] = ['r'],
-        default_device: str = 'cpu',
-        jit: bool = True,
-        *args,
-        **kwargs
-    ):
-        super().__init__(*args, **kwargs)
-        self.device = default_device
-        self.model, _ = clip.load(model_name, self.device, jit)
-        self.default_traversal_paths = default_traversal_paths
-        self.default_batch_size = default_batch_size
-
-    @requests
-    def encode(self, docs: Optional[DocumentArray], parameters: dict, **kwargs):
-        """
-        Encode all docs with text and store the encodings in the embedding
-        attribute of the docs.
-
-        :param docs: documents sent to the encoder. The docs must have text.
-        :param parameters: dictionary to define the ``traversal_path`` and the
-            ``batch_size``. For example,
-            ``parameters={'traversal_paths': ['r'], 'batch_size': 10}``
-        """
-        if docs:
-            document_batches_generator = get_docs_batch_generator(
-                docs,
-                traversal_path=parameters.get(
-                    'traversal_paths', self.default_traversal_paths
-                ),
-                batch_size=parameters.get('batch_size', self.default_batch_size),
-                needs_attr='text',
-            )
-            self._create_embeddings(document_batches_generator)
-
-    def _create_embeddings(self, document_batches_generator: Iterable):
-        with torch.no_grad():
-            for document_batch in document_batches_generator:
-                text_batch = [d.text for d in document_batch]
-                tensor = clip.tokenize(text_batch).to(self.device)
-                embedding_batch = self.model.encode_text(tensor)
-                numpy_embedding_batch = embedding_batch.cpu().numpy()
-                for document, numpy_embedding in zip(
-                    document_batch, numpy_embedding_batch
-                ):
-                    document.embedding = numpy_embedding
+import os
+from typing import Dict, List, Optional
+
+import torch
+from jina import DocumentArray, Executor, requests
+from jina.logging.logger import JinaLogger
+from jina_commons.batching import get_docs_batch_generator
+from transformers import CLIPTokenizer, CLIPModel
+
+
+class CLIPTextEncoder(Executor):
+    """Encode text into embeddings using a CLIP model.
+
+    :param pretrained_model_name_or_path: Can be either:
+        - A string, the model id of a pretrained CLIP model hosted
+            inside a model repo on huggingface.co, e.g., 'openai/clip-vit-base-patch32'
+        - A path to a directory containing model weights saved, e.g., ./my_model_directory/
+        - A path or url to a tensorflow index checkpoint file, e.g, ./tf_model/model.ckpt.index
+        See all CLIP models at https://huggingface.co/models?filter=clip
+    :param base_tokenizer_model: Base tokenizer model.
+        Defaults to `pretrained_model_name_or_path` if None

```

Comment:
```suggestion
        Defaults to ``pretrained_model_name_or_path`` if None
```
Double ticks because this is reST syntax